vllm/model_executor/layers/fused_moe/modular_kernel.py

```diff
@@ -716,13 +716,22 @@ def __init__(
         fused_experts: FusedMoEPermuteExpertsUnpermute,
         shared_experts: torch.nn.Module | None = None,
         shared_experts_stream: torch.cuda.Stream | None = None,
+        parallel_config: ParallelConfig | None = None,
```

Comment:
This should really take a `FusedMoEParallelConfig` instead of a `ParallelConfig`. It has all the necessary info and is always available at any of the construction sites of a modular kernel.